pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/LuceneTextIndexUtils.java

```diff
@@ -33,10 +38,54 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
+/**
+ * Utility class for Lucene text index operations.
+ * Contains common methods for parsing options and creating query parsers.
+ */
 public class LuceneTextIndexUtils {
   private static final Logger LOGGER = LoggerFactory.getLogger(LuceneTextIndexUtils.class);
 
+  // Parser option key
+  public static final String PARSER_OPTION = "parser";
+
+  // Parser values
+  public static final String PARSER_CLASSIC = "CLASSIC";
+  public static final String PARSER_STANDARD = "STANDARD";
+  public static final String PARSER_COMPLEX = "COMPLEX";
+
+  // Default operator values
+  public static final String DEFAULT_OPERATOR_AND = "AND";
+  public static final String DEFAULT_OPERATOR_OR = "OR";
+
+  // Boolean values
+  public static final String TRUE = "true";
+  public static final String FALSE = "false";
+
+  // Option keys - Lucene QueryParser configuration options
+  // These options are applied to Lucene QueryParser instances via reflection
+  // Documentation:
+  // https://lucene.apache.org/core/9_8_0/queryparser/org/apache/lucene/queryparser/classic/QueryParser.html
+  public static final String DEFAULT_OPERATOR_OPTION = "DefaultOperator";
```

Comment:
(minor)
1. Use lower case for the first character
2. We can put these option keys under a subclass `OptionKey`, and remove `OPTION` from the name